src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -316,8 +316,7 @@ protected override void BeginWriteAttribute(CodeRenderingContext context, string
                 .WriteStartMethodInvocation($"{_scopeStack.BuilderVarName}.{nameof(ComponentsApi.RenderTreeBuilder.AddAttribute)}")
                 .Write("-1")
                 .WriteParameterSeparator()
-                .WriteStringLiteral(key)
-                .WriteParameterSeparator();
+                .WriteStringLiteral(key);
```

Comment:
What does removing `WriteParameterSeparator` do here functionally?

Reply:
Writes out `", "`

Reply:
Prevents writing out `", "` which means this method no longer assumes there's additional parameters. Aka imagine going from `AddAttribute(-1, def` to `AddAttribute(-1, abc`

Reply:
@SteveSandersonMS all this is an orthogonal fix no?

Reply:
Technically it could be treated as orthogonal. However the motivating use case for the optimisation is this PR, ie that it will become a much more common code path that now warrants this small bit of streamlining.